monai/data/image_reader.py

```diff
@@ -626,3 +630,143 @@ def _get_spatial_shape(self, img) -> np.ndarray:
         """
         # the img data should have no channel dim or the last dim is channel
         return np.asarray((img.width, img.height))
+
+
+class WSIReader(ImageReader):
+    """
+    Read whole slide imaging and extract patches
+
+    """
+
+    def __init__(self, wsi_reader_name: str = "CuImage"):
+        super().__init__()
+        self.wsi_reader_name = wsi_reader_name.lower()
+        if self.wsi_reader_name == "openslide":
+            self.wsi_reader = openslide.OpenSlide
+            print("> OpenSlide is being used.")
+        elif self.wsi_reader_name == "cuclaraimage":
+            self.wsi_reader = cuimage.CuImage
+            print("> CuImage is being used.")
+        else:
+            raise ValueError('`wsi_reader_name` should be either "CuClaraImage" or "OpenSlide"')
+
+    def verify_suffix(self, filename: Union[Sequence[str], str]) -> bool:
+        """
+        Verify whether the specified file or files format is supported by WSI reader.
+
+        Args:
+            filename: file name or a list of file names to read.
+                if a list of files, verify all the suffixes.
+        """
+        return is_supported_format(filename, ["tif", "tiff"])
+
+    def read(self, data: Union[Sequence[str], str, np.ndarray], **kwargs):
+        """
+        Read image data from specified file or files.
+        Note that the returned object is CuImage or list of CuImage objects.
+
+        Args:
+            data: file name or a list of file names to read.
+
+        """
+        img_: List = []
+
+        filenames: Sequence[str] = ensure_tuple(data)
+        for name in filenames:
+            img = self.wsi_reader(name)
+            if self.wsi_reader_name == "openslide":
+                img.shape = (img.dimensions[1], img.dimensions[0], 3)
+            img_.append(img)
+
+        return img_ if len(filenames) > 1 else img_[0]
+
+    def get_data(
+        self,
+        img_obj,
+        location: Tuple[int, int] = (0, 0),
+        size: Optional[Tuple[int, int]] = None,
```

Comment:
Suggest to use `spatial_size`, otherwise, it may be confusing whether it contains `channel` dim.

Reply:
I kind of agree here but since both OpenSlide and cuClaraImage use `size` to define this input argument, I found using another name confusing for people who has ML in pathology background and has worked with WSI in the past.